CMakeLists.txt

```diff
@@ -9,19 +9,52 @@ if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   find_package(LLVM CONFIG REQUIRED)
   find_package(Clang CONFIG REQUIRED)
 
+  get_target_property(CLANG_BIN clang LOCATION)
+  execute_process(COMMAND ${CLANG_BIN} -print-resource-dir
+                  OUTPUT_VARIABLE clang_default_resource_dir
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+
   list(APPEND CMAKE_MODULE_PATH ${LLVM_DIR})
   include(AddLLVM)
   include(HandleLLVMOptions)
 else()
+  if (CLANG_RESOURCE_DIR)
```

Comment:
Do you need to check if it's set? If not, `clang_default_resource_dir` won't be set and will evaluate to empty, right? So you might as well just set it unconditionally. YOLO!

Reply:
True. I guess the proper default in this case would be would be to clobber together the same path the clang toolchain uses, relative to the clang binary?

Reply:
Do we really have to do anything about this case? If CLANG_RESOURCE_DIR is unset, clang tooling will use the default lookup path to find resources, if it _is_ set, clang tooling will use the override. So if we do nothing, that should just work.

Hence, why not handle in-tree builds by _not_ setting the resource dir?

Reply:
Phew, I feel better now :)

We'd probably need to short-circuit the resource-dir handling in IWYU, then?

And I think there should be a target dependency on `clang-headers` for in-tree, so that building IWYU will automatically get the headers in place. But maybe that's a next step.